### PR TITLE
Add include locale files for common fields

### PR DESCRIPTION
**Pull request by Wolvverine**
from GENERICOBJECT_LOCALES_PATH/field.[language].php

### CODE REVIEW

**Comment by Wolvverine** [comment]
Please to release for both GLPI 10.x.x and for 9.5.x

**Review by stonebuzz — CHANGES_REQUESTED** [review]
Unless I am wrong but
locales fields are already include when ```itemtype``` is registred

https://github.com/stonebuzz/genericobject/blob/80a8e762e9bce97588da893a185499ceee4a4127/inc/field.class.php#L49

What is the original problem?

**Comment by Wolvverine** [comment]
Yes , but only fields for specific itemtype. Not for common fields defined in GENERICOBJECT_FIELDS_PATH/field.constant.php

**Comment by Wolvverine** [comment]
@stonebuzz @cedric-anne 

Do you need any more data?

On another note, there is no option for generic plugin to report errors, and there are quite a few in the plugin.

**Comment by Wolvverine** [comment]
Yes.

**Comment by stonebuzz** [comment]
You do not need to include these locales, they are provided as standard by GLPI.

This will not work
https://github.com/pluginsGLPI/genericobject/blob/a5e5443f25c30ce631b58146969a66f71d2590b0/setup.php#L261-L289

Because for native fields defined in ```field.constant.php```, ```genericobject``` use GLPI native locales ex: ```__("Name")```.
Also ```field.constant.php``` do not have a locale files as ```field.en_GB.php```

Do you want to include locale for any fields that you have added to this file ?

Best regards

**Comment by Wolvverine** [comment]
Not fields defined in the plugin code, but new global fields created for all new assets.
declared in : files/_plugins/genericobject/fields//field.constant.php

image

**Comment by stonebuzz** [comment]
> Not fields defined in the plugin code, but new global fields created for all new assets. declared in : files/_plugins/genericobject/fields//field.constant.php
> 

Ok, i see.

Let me check

File: setup.php [hunk 1]
```diff
@@ -248,6 +248,8 @@ function plugin_genericobject_includeCommonFields($force = false) {
 
    //Include user constants, that must be accessible for all itemtypes
    if (file_exists(GENERICOBJECT_FIELDS_PATH . "/field.constant.php")) {
+      // Load locales
+      plugin_genericobject_includeCommonFieldsLocale($force);
```
Comment (stonebuzz) [inline setup.php:252]:
```suggestion
      plugin_genericobject_includeLocalesGlobalFieldsIfNeeded($force);
```

File: setup.php [hunk 2]
```diff
@@ -256,6 +258,36 @@ function plugin_genericobject_includeCommonFields($force = false) {
    }
 }
 
+
+function plugin_genericobject_includeCommonFieldsLocale($force = false) {
```
Comment (stonebuzz) [inline setup.php:262]:
```suggestion
function plugin_genericobject_includeLocalesGlobalFieldsIfNeeded($force = false) {
```

File: setup.php [hunk 3]
```diff
@@ -256,6 +258,36 @@ function plugin_genericobject_includeCommonFields($force = false) {
...
+   global $CFG_GLPI,$LANG;
```
Comment (stonebuzz) [inline setup.php:263]:
```suggestion
```

File: setup.php [hunk 4]
```diff
@@ -256,6 +258,36 @@ function plugin_genericobject_includeCommonFields($force = false) {
...
+   //Include user constants locales accessible for all itemtypes
+
+   if (isset ($_SESSION["glpilanguage"])) {
+      $locale_common_file ='field' . '.' . $_SESSION['glpilanguage'];
```
Comment (stonebuzz) [inline setup.php:267]:
```suggestion
      $locale_global_file ='field' . '.' . $_SESSION['glpilanguage'];
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Comment by cedric-anne** [comment]
@Wolvverine

I do not know well how to customize this plugin. Can you attach here custom files required to test this feature ?

**Comment by Wolvverine** [comment]
Standard way for generic object, example:
image

files/_plugins/genericobject/fields//field.constant.php - file with common fields:

```
<?php
global $GO_FIELDS, $LANG;

$GO_FIELDS['plugin_genericobject_orgunits_id']['name']       = $LANG['genericobject']['CommonFields'][1];
$GO_FIELDS['plugin_genericobject_orgunits_id']['field']      = 'Organizational Unit';
$GO_FIELDS['plugin_genericobject_orgunits_id']['input_type'] = 'dropdown';

$GO_FIELDS['idfield']['name'] =  $LANG['genericobject']['CommonFields'][2];
$GO_FIELDS['idfield']['input_type'] = 'text';

?>
```

in locales files:
files/_plugins/genericobject/locales/field.en_GB.php
files/_plugins/genericobject/locales/field.en_US.php

```
<?php
// Fields
$LANG['genericobject']['CommonFields'][1]="Organizational Unit"; 
$LANG['genericobject']['CommonFields'][2]="ID";
```

File: setup.php (hunk 3 above)
Comment (Wolvverine) [reply setup.php:263]: I can't remember at the moment, but I don't think it works properly on my installations without it.

**Review by stonebuzz — APPROVED** [review]

**Comment by cedric-anne** [comment]
Refactored and cherry-picked in 25d7b1cee9c9f8faf1ef9b253aba23c8cd71547a.

I changed filename from `field.$lang.php` to `fields.$lang.php`.

**Comment by cedric-anne** [comment]
Documentation update: https://github.com/pluginsGLPI/doc/pull/61

**Pull request closed without merging**